### ISSUE TITLE
Refactor prep_riboviz.py logging to produce step-specific log files

### ISSUE THREAD

**Issue by mikej888** [riboviz/RiboViz#55]
Update prep_riboviz.py logging:

* Each third-party tool invocation has a separate log file.
* Each log file is prefixed by sample name and the index of the step in the workflow.
* prep_riboviz.py has its own log file too, which records the names of each of the other log files created.
* Log files for a run are put in a log file directory.
* The log file directory, or files within, should have timestamps in their names also.

**Comment by mikej888** [comment]
See [prep-riboviz-log-files-55](https://github.com/riboviz/RiboViz/tree/prep-riboviz-log-files-55) branch.

prep_riboviz.py now creates log files for each of the commands it invokes in a timestamped directory under vignette/logs/. The location of these files is configurable via a dir_logs parameter in YAML.

prep_riboviz.py itself creates a timestamped log file in the current directory e.g. riboviz-20190919-064919-968064.log. Information on key steps is still printed to the command-line too.

The log files means there's no need to do output redirection when running prep_riboviz.py. It can now be run as:

```console
python -m riboviz.tools.prep_riboviz rscripts/ vignette/vignette_config.yaml
```
```console
PYTHONPATH=. python riboviz/tools/prep_riboviz.py rscripts/ vignette/vignette_config.yaml
```

doc/run-vignette.md describes the log files output in more detail.

prep_riboviz.py now uses FileNotFoundError for more effective reporting of missing files. This is a Python 3 only exception (see #54)

**Comment by mikej888** [comment]
Version before merge into develop branch: c5c6b6193305230b5c6f8c0fba7a0f91d3548034

**Comment by mikej888** [comment]
Merged as part of pull requests #70 and #71.
